File: TwoPointers/csharp/Uncompress.cs
using System.Text;

namespace TwoPointersSolutions;

public static class Uncompress
{
    public static string Run(string s)
    {
        var builder = new StringBuilder();
        var i = 0;
        while (i < s.Length)
        {
            var j = i;
            while (j < s.Length && char.IsDigit(s[j]))
            {
                j++;
            }

            var count = int.Parse(s[i..j]);
            var ch = s[j];
            builder.Append(new string(ch, count));
            i = j + 1;
        }

        return builder.ToString();
    }
}

File: TwoPointers/java/Uncompress.java
public final class Uncompress {
  private Uncompress() {
  }

  public static String run(String s) {
    StringBuilder builder = new StringBuilder();
    int i = 0;

    while (i < s.length()) {
      int j = i;
      while (j < s.length() && Character.isDigit(s.charAt(j))) {
        j++;
      }
      int count = Integer.parseInt(s.substring(i, j));
      char ch = s.charAt(j);
      builder.append(String.valueOf(ch).repeat(count));
      i = j + 1;
    }

    return builder.toString();
  }
}
